File: src/Wallet.Entities/EntityObjects/Message.cs
using Wallet.Entities.Base.Concrete;

namespace Wallet.Entities.EntityObjects;

public class Message : SoftDeleteEntity
{
    public Guid SenderId { get; set; }
    public User? Sender { get; set; }
    
    public Guid ReceiverId { get; set; }
    public User? Receiver { get; set; }
    
    public required string Subject { get; set; }
    public required string Content { get; set; }
    public bool IsRead { get; set; }
    public DateTime? ReadAt { get; set; }
    
    // Yanıt zinciri için
    public Guid? ParentMessageId { get; set; }
    public Message? ParentMessage { get; set; }
    public ICollection<Message>? Replies { get; set; }
} 

File: src/wallet.ui/src/types/message.ts
export interface IMessage {
  id: string;
  senderId: string;
  senderUsername: string;
  senderFullName: string;
  receiverId: string;
  receiverFullName: string;
  subject: string;
  content: string;
  isRead: boolean;
  readAt: string | null;
  createdDate: string;
  parentMessageId: string | null;
  replies?: IMessage[];
}

export interface IMessageCreate {
  receiverUsername: string;
  subject: string;
  content: string;
  parentMessageId?: string;
} 